pyvista/plotting/plotting.py

```diff
@@ -3836,3 +3834,98 @@ def plot(self, *args, **kwargs):
+            ``as_global=False`` will cause the background to only
+            appear in one window.
+
+        Examples
+        --------
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> plotter = pyvista.Plotter()
+        >>> actor = plotter.add_mesh(pyvista.Sphere())
+        >>> plotter.add_background_image(examples.mapfile)
+        >>> plotter.show() # doctest:+SKIP
+
+        """
+        # make a renderer as layer zero and move all other renders forward
+        self._background_renderer = vtk.vtkRenderer()
+        self._background_renderer.SetLayer(0)
+        self._background_renderer.InteractiveOff()
+        self._background_renderer.SetBackground(self.renderer.GetBackground())
+        self.ren_win.SetNumberOfLayers(len(self.renderers) + 1)
+        self.ren_win.AddRenderer(self._background_renderer)
+
+        if as_global:
+            for renderer in self.renderers:
+                renderer.SetLayer(1)
+                self.ren_win.AddRenderer(renderer)
+        else:
+            view_port = self.renderer.GetViewport()
+            self._background_renderer.SetViewport(view_port)
+            self.renderer.SetLayer(1)
+
+        image_data = pyvista.read(image_path)
+
+        image_actor = vtk.vtkImageActor()
+        image_actor.SetInputData(image_data)
+        self._background_renderer.AddActor(image_actor)
+        self._background_camera = self._background_renderer.GetActiveCamera()
+        self._background_camera.ParallelProjectionOn()
+
+        origin = image_data.GetOrigin()
+        extent = image_data.GetExtent()
+        spacing = image_data.GetSpacing()
+        xc = origin[0] + 0.5*(extent[0] + extent[1]) * spacing[0]
+        yc = origin[1] + 0.5*(extent[2] + extent[3]) * spacing[1]
+        yd = (extent[3] - extent[2] + 1) * spacing[1]
+        d = self._background_camera.GetDistance()
+
+        # make the longest dimensions match the plotting window
+        img_dim = np.array(image_data.dimensions[:2])
+        self._background_camera.SetFocalPoint(xc, yc, 0.0)
+        self._background_camera.SetPosition(xc, yc, d)
+
+        def resize_background():  # pragma: no cover
+            ratio = img_dim/np.array(self.window_size)
+            scale_value = 1
+            if ratio.max() > 1:
+                # images are not scaled if larger than the window
+                scale_value = ratio.max()
+
+            if scale is not None:
+                scale_value /= scale
+
+            self._background_camera.SetParallelScale(0.5 * yd * scale_value)
+
+        resize_background()
+
+        # setup autoscaling of the image
+        if auto_resize and hasattr(self, 'iren'):  # pragma: no cover
+            def on_resize(*args):
+                if self.__prior_window_size != self.window_size:
+                    self.__prior_window_size = self.window_size
+                    resize_background()
+
+            self.iren.AddObserver('ModifiedEvent', on_resize)
```

Comment:
We should use our brand new `_add_observer` here.

Reply:
Now using it as of e1f848b307f08cdf03442cc02911ffead8a2945a.